app/src/main/java/com/hover/stax/transfers/NonTemplateSummaryAdapter.kt

```diff
@@ -0,0 +1,40 @@
+package com.hover.stax.transfers
+
+import android.annotation.SuppressLint
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.hover.stax.databinding.SummarycardNontemplateItemsBinding
+
+class NonTemplateSummaryAdapter : RecyclerView.Adapter<NonTemplateSummaryAdapter.ViewHolder>() {
+    private var itemMap = mutableMapOf<String, String>()
+
+    @SuppressLint("NotifyDataSetChanged")
+    fun updateList(key: String, value: String) {
+        itemMap.put(key, value)
+        notifyDataSetChanged()
+    }
+
+    inner class ViewHolder(val binding: SummarycardNontemplateItemsBinding) : RecyclerView.ViewHolder(binding.root) {
+        fun bindItems(key: String, value: String) {
+            binding.itemKey.text = key
```

Comment:
Since the `itemKey` is not an inflatable view, it doesn't need to be part of the view/layout. If the `key` needs to be accessed, then the property in the `bindItems` function is enough. .

Reply:
Actually, the itemKey is an inflatable view, however it's badly named. I've changed the word to be itemLabel, not key.

Reply:
Done